### ISSUE TITLE
💡 RFC: Support AsciiDoc for technical writing

### ISSUE THREAD

**Issue by yonkeltron** [snowpackjs/astro#1013]
### Background & Motivation

I want to be able to use a document format more suitable for technical writing. For me and my team, this is [AsciiDoc](https://asciidoctor.org/docs/what-is-asciidoc/) as it supports everything that Markdown supports and considerably more, without much additional complexity. I also find it enjoyable to write with a regular syntax and relatively few surprises. While heavier solutions exist, such as [Antora](https://antora.org/), I'd like to be able to author technical content with Astro.

A `<AsciiDoc>` component would be ideal, just like the `<Markdown>` component in `astro/components`.

### Possible solutions

AsciiDoctor is now the official toolchain for converting AsciiDoc and supports things beautifully. While it is written in Ruby, it has been cross-compiled to JavaScript with [Opal](https://opalrb.com/) thus yielding [Asciidoctor.js](https://github.com/asciidoctor/asciidoctor.js). Both the original Ruby implementation and the JavaScript port are official, mature, and maintained projects.

### Alternatives considered

Since build-time is where the rendering would happen, Astro could call out to either the Ruby version of AsciiDoctor, or an alternative implementation such as [Pandoc](https://pandoc.org/). However: the latter, Pandoc option is not as featureful.

### Risks, downsides, and/or tradeoffs

There is reason to be concerned about growing complexity and/or bloat, I suppose. However since the focus of Astro is content, another great way to author content will only strengthen the value prop.

### Open Questions

* Should this be done at all?
* Should this be done in the core product?
* Should this be done in some sort of a plugin?

### Help make it happen!

- [ ] I am willing to submit a PR to implement this change.
- [X] I am willing to submit a PR to implement this change, but would need some guidance.
- [ ] I am not willing to submit a PR to implement this change.

**Comment by drwpow** [comment]
RFC Call decision: this would be a really neat idea as a third-party Astro component, rather than core. If this could be implemented externally, then it would really open some barriers to other languages being supported as well, such as [LaTeX](https://www.latex-project.org/).

Some people expressed a lot of interest in helping develop this! So to that end we’ll start a thread in #ideas-and-suggestions in Discord to discuss further:

And in the meantime, if anyone would like to work on this and post it to [awesome-astro](https://github.com/one-aalam/awesome-astro), that’d be appreciated as well.

**Comment by shishkin** [comment]
Could someone provide some guidance what would it take to support AsciiDoc in an external integration? I saw a the [astro-adoc component](https://github.com/devidw/astro-addons/tree/main/packages/astro-adoc) that does what the original RFC suggested. I would like however to be able to write pages in AsciiDoc alongside `.astro` and `.md`. Would Astro just magically pick up `.adoc` pages with [vite-plugin-asciidoc](https://github.com/Djaler/vite-plugin-asciidoc) installed or there something else that the integration need to do?

**Comment by shishkin** [comment]
I've experimented with the idea and after some reading of astro markdown integration code and trial and error I was able to render AsciiDoc pages. I had to use some undocumented integration API and this is my first attempt at vite plugins and astro rendering pipeline. In any case, I've release my experiment as an [NPM package](https://www.npmjs.com/package/astro-asciidoc). If anybody is interested I'd be grateful for testing and ideas how to improve the integration. The [project is here](https://github.com/shishkin/astro-asciidoc).
